src/thread.rs

```diff
@@ -163,43 +169,73 @@ where
 ///
 /// ```
 /// crossbeam_utils::thread::scope(|scope| {
-///     scope.spawn(|| println!("Exiting scope"));
-///     scope.spawn(|| println!("Running child thread in scope"));
+///     scope.spawn(|_| println!("Exiting scope"));
+///     scope.spawn(|_| println!("Running child thread in scope"));
 /// }).unwrap();
 /// ```
 pub fn scope<'env, F, R>(f: F) -> thread::Result<R>
 where
     F: FnOnce(&Scope<'env>) -> R,
 {
+    let (tx, rx) = mpsc::channel();
```

Comment:
Here we're using mpsc as a barrier, right?  If so, I'd like to use: https://doc.rust-lang.org/std/sync/struct.Barrier.html

Reply:
We can't use `Barrier` because `Barrier::new(n)` must know in advance how many threads are going to be synchronized (that's the `n`).

What we need here is [`WaitGroup`](https://docs.rs/chan/0.1.23/chan/struct.WaitGroup.html), but I'd rather not add `chan` as a dependency because it's deprecated.

I'd like to move `WaitGroup` into crossbeam in an another PR and then replace this use of `mpsc` with it.

Reply:
Now we have `WaitGroup` :)